paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/PrimaryKeyLookupTable.java

```diff
@@ -99,25 +99,35 @@ public void refresh(Iterator<InternalRow> incremental) throws IOException {
         Predicate predicate = projectedPredicate();
         while (incremental.hasNext()) {
             InternalRow row = incremental.next();
-            primaryKeyRow.replaceRow(row);
-            if (userDefinedSeqComparator != null) {
-                InternalRow previous = tableState.get(primaryKeyRow);
-                if (previous != null && userDefinedSeqComparator.compare(previous, row) > 0) {
-                    continue;
+            if (refreshAsync) {
```

Comment:
If this is a common piece of code, I think it could be implemented in `FullCacheLookupTable` and then provide an abstract `refreshRow` interface to subclasses.

Reply:
DONE